src/entry_point.rs

```diff
@@ -254,7 +254,7 @@ pub unsafe extern "C" fn _start() -> ! {
     mv   a0, s0             // first arg is app_start
     mv   s0, sp             // Set the frame pointer to sp.
     mv   a1, s1             // second arg is stacktop
-    mv   a1, s1             // second arg is stackto
+    mv   a2, s1             // third arg is app_heap_break
```

Comment:
Shouldn't the source be `s2`?

Reply:
Good catch!
It shouldn't be s2, that is some leftover value. It also shouldn't be s1 though.
It should be a3, assuming that it won't be clobbered by the ecalls in the future or a3 should be saved somewhere (like s2).
Unfortunately changing this breaks the app, so I will have to look into that. I'm going to drop this patch and we can revisit in the future.